(Legacy) develop_FW-866 OpenRadio interface

minarady1 opened the pull request:
Initial commit of code compiling and working and synching but not fully communicating

minarady1 committed:
Initial commit of code compiling and working and synching but not fully communicating

minarady1 committed:
wrote radio_set_modulation and radio_local_bootstrap

minarady1 committed:
some updates

minarady1 committed:
open radio working :)

minarady1 committed:
some updates, mainly trying to debug

minarady1 committed:
updates

minarady1 committed:
radio at86rf files

minarady1 committed:
Integrating cc2538rf into the open radio interface

minarady1 committed:
using openradio in openwsn

minarady1 committed:
open radio interface finally working and integrated into openwsn

minarady1 committed:
full radio set added to openradio

minarady1 committed:
Complete open radio interface for openmote-b tested with 7 modulations.

minarady1 committed:
Restoring default configuration

minarady1 committed:
restoring default config

minarady1 committed:
resorting default configurations and removing unnecessary code used for debugging

minarady1 committed:
restoring bsp_radio_tx default code

minarady1 committed:
restoring sniffer code

minarady1 committed:
restoring sniffer code

minarady1 committed:
restoring sniffer code

minarady1 committed:
restoring bsp_radio_rx code

minarady1 committed:
A project for demonstrating the open radio interface on openmote-b:
- A hybrid transmitter for openmote-b which loops on all available modulations using openradio interface.
- A receiver which sets the rx settings also using open radio interface

minarady1 committed:
Merge branch 'develop_FW-866' of https://github.com/minarady1/openwsn-fw into develop_FW-866

minarady1 committed:
openradio interface cleaning in progress

minarady1 committed:
modivided slot 40 for FSK and added slot 41 for ofdm1 mcs3

minarady1 committed:
introduced slot_template instead of slot duration
cleaning radio header and c file

minarady1 committed:
cleaneing openradio interface + adding comments + using slot template indexes instead of slot durations

minarady1 committed:
comments+cleaning

minarady1 committed:
Merge branch 'develop' into develop_FW-866

minarady1 committed:
minor fixes

minarady1 committed:
macpong

minarady1 committed:
Merge branch 'develop_FW-866' of https://github.com/minarady1/openwsn-fw into develop_FW-866

minarady1 committed:
changing slot templates to struct variables instead of #def constants

fjmolinas commented:
Really like the struct based approach for `slot` information!

fjmolinas commented on bsp/boards/openmote-b/board_info.h:
```diff
@@ -75,92 +75,33 @@
+    // also implementation related (because the time to execute the Tx/Rx command is highly dependent on the radio AND the configuration)
+    uint8_t PORT_delayTx;                       
+    uint8_t PORT_delayRx;
+} slot_board_vars_t; //board specific slot vars
```
Does it make sense defining this in `bsp/boards/board.h`? you could have something
like this as a default config:

```c

typedef struct {
    uint8_t SLOTDURATION;
    uint8_t PORT_TsSlotDuration;
    uint8_t PORT_maxTxDataPrepare;
    uint8_t PORT_maxRxAckPrepare;
    uint8_t PORT_maxRxDataPrepare;
    uint8_t PORT_maxTxAckPrepare;
    uint8_t PORT_delayTx;
    uint8_t PORT_delayRx;
} slot_board_vars_t;

/* This would be a default configuration */

#ifndef 10MS_SLOTDURATION
#define 10MS_SLOTDURATION             20   /* in miliseconds */
#endif
#ifndef 10MS_PORT_TsSlotDuration
#define 10MS_PORT_TsSlotDuration     655   /* 655 ticks at @32768Hz */
#endif
#ifndef 10MS_PORT_maxTxDataPrepare
#define 10MS_PORT_maxTxDataPrepare   110   /* 3355us (not measured) */
#endif
#ifndef 10MS_PORT_maxRxAckPrepare
#define 10MS_PORT_maxRxAckPrepare    20    /*  610us (not measured) */
#endif
#ifndef 10MS_PORT_maxRxDataPrepare
#define 10MS_PORT_maxRxDataPrepare   33    /* 1000us (not measured) */
#endif
#ifndef 10MS_PORT_maxTxAckPrepare
#define 10MS_PORT_maxTxAckPrepare    50    /* 1525us (not measured) */
#endif
#ifndef 10MS_PORT_delayTx
#define 10MS_PORT_delayTx            10   /*  549us (not measured) */
#endif
#ifndef 10MS_PORT_delayRx
#define 10MS_PORT_delayRx            0    /*    0us (can not measure) */
#endif

#define SLOT_10_MS_BOARD_CONFIG_DEFAULT  { .SLOTDURATION = 10MS_SLOTDURATION, \
                                           .PORT_TsSlotDuration = 10MS_PORT_TsSlotDuration, \
                                           .PORT_maxTxDataPrepare = 10MS_PORT_maxTxDataPrepare, \
                                           .PORT_maxRxAckPrepare = 10MS_PORT_maxRxAckPrepare, \
                                           .PORT_maxRxDataPrepare = 10MS_PORT_maxRxDataPrepare, \
                                           .PORT_maxTxAckPrepare = 10MS_PORT_maxTxAckPrepare, \
                                           .PORT_delayTx = 10MS_PORT_delayTx, \
                                           .PORT_delayRx = 10MS_PORT_delayRx, \
                                           }

static const slot_board_vars_t slot_10ms_board_vars[] =
{
#ifdef SLOT_10MS_BOARD_VARS
    SLOT_10MS_BOARD_VARS,
#else
    SLOT_10MS_BOARD_VARS_DEFAULT,
#endif
...
/* that for all slots */
...
#ifdef SLOT_40MS_OFDM1MCS0_3_SUBGHZ_BOARD_VARS
    SLOT_40MS_OFDM1MCS0_3_SUBGHZ_BOARD_VARS,
#else
    SLOT_40MS_OFDM1MCS0_3_SUBGHZ_BOARD_VARS_DEFAULT,
#endif
};
```

And then per `board_info` you would keep the original defines and do the following:

```c
#define SLOT_10MS_BOARD_VARS  { .SLOTDURATION = 10MS_SLOTDURATION, \
                             .PORT_TsSlotDuration =10MS_ PORT_TsSlotDuration, \
                             .PORT_maxTxDataPrepare = 10MS_PORT_maxTxDataPrepare, \
                             .PORT_maxRxAckPrepare =10MS_ PORT_maxRxAckPrepare, \
                             .PORT_maxRxDataPrepare = 10MS_PORT_maxRxDataPrepare, \
                             .PORT_maxTxAckPrepare = 10MS_PORT_maxTxAckPrepare, \
                             .PORT_delayTx = 10MS_PORT_delayTx, \
                             .PORT_delayRx = 10MS_PORT_delayRx, \
                             }
```

I think this would also make the migration easier and the init function could
simply be:

```c
void board_init_slot_vars(const slot_board_config_t *slot_board_config) {
    slot_board_vars = *slot_board_config;
}
```

The nice thing is this change could be split from this PR, and since currently only one `SLOTDURATION` is possible there would be no need to change any define per `BOARD_INFO` just add:

```
```c
#define SLOT_BOARD_VARS  { .SLOTDURATION = SLOTDURATION, \
                             .PORT_TsSlotDuration =PORT_TsSlotDuration, \
                             .PORT_maxTxDataPrepare = PORT_maxTxDataPrepare, \
                             .PORT_maxRxAckPrepare = PORT_maxRxAckPrepare, \
                             .PORT_maxRxDataPrepare = PORT_maxRxDataPrepare, \
                             .PORT_maxTxAckPrepare = PORT_maxTxAckPrepare, \
                             .PORT_delayTx = PORT_delayTx, \
                             .PORT_delayRx = PORT_delayRx, \
                             }
```

minarady1 committed:
slot_duration struct update

minarady1 committed:
adding few helper functions to handle slot info

minarady1 committed:
quick update

minarady1 committed:
fixed dynamic slot selection + code formatting

minarady1 committed:
Final fixes to the dynamic slot selection at runtime. This feature now works correctly :)

- moved time_type_t to board.h because this enum seems fundamental to the firmware (even lower than opentimers where it used to be define). It is currently used in a new function in board.c: board_getSlotDuration (time_type_t)

- MACRO PRE_CALL_TIMER_WINDOW in opentimers.h now refers to function call keeping track of the current slot board_getSlotDuration(TIME_TICS) instead of the old constant PORT_TsSlotDuration

- Added a double radio pin toggle in ri4 and ri5 to indicate the actual start and end of frame at LA for accurate measurement of synchronization offset

- both sixtop.c and icmpv6rpl.c now refer to function call keeping track of the current slot board_getSlotDuration(TIME_MS) instead of the old constant: SLOTDURATION

minarady1 committed:
cleaning up code for bsp, projacts, MAC, IAR project files and .gitignore

minarady1 committed:
Updated radio drivers for python board + at86rf231 board to be compliant with the new radio.h interface
Updated board-specific definitions to be compliant to the new 802.154.e dynamic slot/radio selection (and to the new board.h). This was completed and compiled locally (using armgcc scons compilers) with the following (travis) commands:

- scons board=openmote-b-24ghz  toolchain=armgcc oos_openwsn
- scons board=openmote-cc2538   toolchain=armgcc oos_openwsn
- scons board=openmote-cc2538   toolchain=armgcc oos_sniffer
- scons board=openmote-cc2538   toolchain=armgcc oos_macpong
- scons board=samr21_xpro       toolchain=armgcc oos_openwsn
- scons board=python            toolchain=gcc    oos_openwsn
- scons board=iot-lab_A8-M3     toolchain=armgcc oos_macpong
- scons board=iot-lab_A8-M3     toolchain=armgcc oos_openwsn
- scons board=iot-lab_M3        toolchain=armgcc oos_macpong
- scons board=iot-lab_M3        toolchain=armgcc oos_openwsn
- scons board=openmotestm       toolchain=armgcc oos_macpong
- scons board=openmotestm       toolchain=armgcc oos_openwsn
- scons board=openmote-b        toolchain=armgcc oos_openwsn

minarady1 commented:
Hello @twatteyne , @changtengfei , and @TimothyClaeys  The PR is now ready for review. To test it, you can use SCons or IAR. Make sure you set the following settings:
- Use `openmote-b` board option
- Used fixed channel (e.g. 11)
- Set `EB_PORTION` to 4
- Set `SLOTFRAME_LENGTH` to at least 19
- select the desired slot template and radio setting to use,  currently, the following slot template/radio setting combination in ieee802.15.4e init function are supported:

        SLOT_20ms_24GHZ              , RADIOSETTING_24GHZ --> Default
        SLOT_40ms_24GHZ              , RADIOSETTING_24GHZ -->Essential
        SLOT_40ms_FSK_SUBGHZ        ,RADIOSETTING_FSK_OPTION1_FEC -->Essential
        SLOT_40ms_OFDM1MCS0_3_SUBGHZ , RADIOSETTING_OFDM_OPTION_1_MCS0
        SLOT_40ms_OFDM1MCS0_3_SUBGHZ , RADIOSETTING_OFDM_OPTION_1_MCS1
        SLOT_40ms_OFDM1MCS0_3_SUBGHZ , RADIOSETTING_OFDM_OPTION_1_MCS2
        SLOT_40ms_OFDM1MCS0_3_SUBGHZ , RADIOSETTING_OFDM_OPTION_1_MCS3 -->Essential

minarady1 committed:
restoring defaults

twatteyne commented:
General comments:
- start all of your commit messages with the key of the issue ti addresses. "FW-866. poipoi"
- have the name of the PR contain both the key and title of the issue it fixes "FW-866. OpenRadio interface"

twatteyne commented on .gitignore:
```diff
@@ -1,6 +1,9 @@
 Debug
 settings
 *.dep
+*.ewd
```
good

twatteyne commented on bsp/boards/board.h:
```diff
@@ -22,16 +22,50 @@ typedef enum {
    KICK_SCHEDULER        = 1,
 } kick_scheduler_t;
 
+typedef enum {
```
wasn't this already defined somewhere?

twatteyne commented on bsp/boards/board.h:
```diff
@@ -22,16 +22,50 @@ typedef enum {
+    // also implementation related (because the time to execute the Tx/Rx command is highly dependent on the radio AND the configuration)
+    uint16_t PORT_delayTx;                       
+    uint16_t PORT_delayRx;
+} slot_board_vars_t; //board specific slot vars
```
looks like this _is_ the timeslot template. Suggest renaming to `timeslot_template_t`

minarady1 replied:
oh no, the timelost template is defined in the ieee154e.h. This struct is strictly for board specific parameters (regardless of which 154e timeslot template is used)

twatteyne commented on bsp/boards/board.h:
```diff
@@ -22,16 +22,50 @@ typedef enum {
+
 //=========================== typedef =========================================
+typedef struct {
+    uint16_t PORT_SLOTDURATION;
```
- no need to keep the prefix `PORT`.
- don't use all caps if now a variable
- why is there both `PORT_SLOTDURATION` and `PORT_TsSlotDuration`?

Suggested names:
- `SlotDuration`
- `slotDuration`
- `maxTxDataPrepare`
etc.

twatteyne commented on bsp/boards/board.h:
```diff
@@ -22,16 +22,50 @@ typedef enum {
+    SLOT_40ms_FSK_SUBGHZ,
+    SLOT_40ms_OFDM1MCS0_3_SUBGHZ,
+    MAX_SLOT_TYPES,
+} slotType_t;
```
I don't understand why the slot duration is in the name. Just have 3 timeslot templates, for the 3 radio settings we care about.

```
// available slot templates
typedef enum{
    TIMESLOT_TEMPLATE_OQPSK_24GHZ,
    TIMESLOT_TEMPLATE_2FSK_SUBGHZ,
    TIMESLOT_TEMPLATE_OFDM_SUBGHZ,
    TIMESLOT_TEMPLATE_MAX
} timeslot_templates_t;
```

minarady1 replied:
There is also a 10ms slot template and 20ms slot template. Thoe former is for python the latter has been classically used for 2.4GHz). 
I didn't touch them because they are out of my scope (I can still rempve them, or give them different names if you wish?)

twatteyne commented on bsp/boards/board.h:
```diff
@@ -22,16 +22,50 @@ typedef enum {
+void              board_init(void);
+void              board_sleep(void);
+void              board_reset(void);
+void              board_init_slot_vars(void);
```
don't understand why this function exists. Remove?

minarady1 replied:
I moved it to the private section, it is necessary to initialize the lookup table for the board_vars relevant to each radioSetting

twatteyne commented on bsp/boards/board.h:
```diff
@@ -22,16 +22,50 @@ typedef enum {
+uint16_t          board_getSlotDuration (time_type_t time_type);
+void              board_setSlotType(slotType_t slot_type);
+slot_board_vars_t board_getSlotTemplate (void);
 /**
```
those 3 functions should be removed. Replace by `board_getTimeslotTemplates`.

minarady1 replied:
Ok, so I combined `board_setSlotType `and `board_getSlotTemplate  ` into `slot_board_vars_t board_selectSlotTemplate (slotType_t slot_type)`. I left `board_getSlotDuration ` as is because it is accessed really frequently by the opentimer (and somewhat frequently by rpl) is that alright?

twatteyne commented on bsp/boards/iot-lab_A8-M3/board.c:
```diff
@@ -96,8 +101,48 @@ void board_init(void){
     //enable nvic for the radio
     NVIC_radio();
+    
+    board_init_slot_vars();
```
don't call another function, fill in the structre here.

fjmolinas replied:
What about having an approach like https://github.com/openwsn-berkeley/openwsn-fw/pull/504#discussion_r410117118? You would need fewer lines of code modifications in every `board_info.h`, and could have a default for all boards which parameters have not been measured, you could initialize the structure outside of `bsp/` as well.

minarady1 replied:
@twatteyne not sure I understand, which structure do you mean? 
@fjmolinas  well, the general idea that slot selection will be done at runtime and the selection is done at MAC level (and the MAC level will have the default selection) but we can discuss more in the meeting to make sure I understood you correctly! Thanks a lot for your feedback!

fjmolinas replied:
@minarady1 I PR'd what I meant to your branch https://github.com/minarady1/openwsn-fw/pull/1. Did it only for 1 board.

twatteyne commented on bsp/boards/iot-lab_A8-M3/board.c:
```diff
@@ -96,8 +101,48 @@ void board_init(void){
+uint16_t board_getSlotDuration (time_type_t time_type)
+{
+  if (time_type == TIME_MS)
+    return slot_board_vars [selected_slot_type].PORT_SLOTDURATION;
```
you cannot keep maintaining 2 variables with 2 units. Kill the "ms" variables, keep only the one in ticks

twatteyne commented on bsp/boards/iot-lab_M3/board.c:
```diff
@@ -19,6 +19,11 @@
 #include "cryptoengine.h"
 
+
+//=========================== variables ============================================
```
these headers should be always the same width:
- `//==` at the beginning
- `==` until column 30, then description
- the `==`  until column 80
- empty afterwards

twatteyne commented on bsp/boards/openmote-b-24ghz/radio.c:
```diff
@@ -45,6 +45,8 @@ typedef struct {
 radio_vars_t radio_vars;
 
+// global radio selection, will use RADIOSETTING_24GHZ by default at initialization. 
+uint8_t SELECTED_RADIO      =       RADIOSETTING_24GHZ;
```
the name `SELECTED_RADIO` suggests it's a `#define`. Not sure what is going on.

minarady1 replied:
fixed it. used to be #define and forgot to change capitalization

twatteyne commented on bsp/boards/openmote-b-subghz/board_info.h:
```diff
@@ -128,7 +129,6 @@ static const uint8_t rreg_uriquery[]        = "h=ucb";
 static const uint8_t infoBoardname[]        = "openmote-b-subghz";
 static const uint8_t infouCName[]           = "CC2538";
 static const uint8_t infoRadioName[]        = "CC2538 SoC";
-
```
make sure you have an empty line around all `====` separators.

twatteyne commented on bsp/boards/openmote-b/board.c:
```diff
@@ -18,7 +18,9 @@
 #include <source/gptimer.h>
 #include <source/sys_ctrl.h>
 
+#include "opentimers.h"
```
No. opentimers is at `driver/` level. You cannot include a driver-level module in a BSP module.

twatteyne commented on bsp/boards/openmote-b/board.c:
```diff
@@ -47,20 +46,22 @@ bool board_timer_expired(uint32_t future);
 static void button_init(void);
-static void antenna_init(void);
 
+static void antenna_init(void);
```
there are 2 antennas, why these functions?!?

minarady1 replied:
I am not really sure. My understanding is the they define which radio chip uses the 24ghz antenna

dbarthel-ol replied:
Indeed, there are two antennas on the OpenMoteB. One antenna is 2.4 GHz only, the other one is 868 MHz only. And there are 3 radios in total, two at 2.4 GHz and one at 868 MHz. On the OpenMoteB schematic, I see U8 (bottom right), which looks like an RF switch. It switches the 2.4 GHz antenna between the cc2538 2.4 GHz radio port and the at86rf215 2.4 GHz radio port. PD3 and PD4 of the cc2538 look like the control signals of this RF switch. The functions discussed here activate these PD3 and PD4 signals. (see lines 104-111 of this file, and lines 75-76 of board_info.h)

twatteyne commented on bsp/boards/openmote-b/board.c:
```diff
@@ -106,10 +125,10 @@ void board_sleep(void) {
  */
 void board_timer_init(void) {
-    // Configure the timer
+    /* Configure the timer */
```
comments are always `//`

minarady1 replied:
I did my best fixing those now., I see this style used many board definitions

twatteyne commented on bsp/boards/openmote-b/board.c:
```diff
@@ -133,17 +153,98 @@ bool board_timer_expired(uint32_t future) {
     uint32_t current;
     int32_t remaining;
 
+    /* Get current time */
```
comments using `//`. Please apply throughout.

twatteyne commented on bsp/boards/openmote-b/board.c:
```diff
@@ -133,17 +153,98 @@ bool board_timer_expired(uint32_t future) {
 
+//==== bootstrapping slot info lookup table
+void board_init_slot_vars(void){
+    //10ms slot
```
we don't care about anything than 3 settings, and 40ms timeslots.

minarady1 replied:
the 10ms and 20ms are used in other board, should I definitely remove them? @changtengfei @TimothyClaeys

twatteyne commented on bsp/boards/openmote-b/board.c:
```diff
@@ -183,9 +278,9 @@ static void clock_init(void) {
+    /* Set the IO clock to operate at 32 MHz */
     /* This way peripherals can run while the system clock is gated */
-    SysCtrlIOClockSet(SYS_CTRL_SYSDIV_16MHZ);
+    SysCtrlIOClockSet(SYS_CTRL_SYSDIV_32MHZ);
```
why did you make this change?

minarady1 replied:
I just  replaced the board.c file with the one from the subghz project by Jonathan since I knew it was working.

twatteyne commented on bsp/boards/openmote-b/board.c:
```diff
@@ -206,25 +301,16 @@ static void clock_init(void) {
 static void button_init(void) {
-    volatile uint32_t i;
-
-    /* Delay to avoid pin floating problems */
```
why remove this?

minarady1 replied:
same

twatteyne commented on bsp/boards/openmote-b/board_info.h:
```diff
@@ -57,41 +57,9 @@
-    // radio watchdog
-#endif
+// Number of available radios
+#define MAX_RADIOS  7
```
no need for this. You have a variable which holds your timeslot templates, and can derive the number of templates from there (by dividing by the length of one.)

minarady1 replied:
this is not for templates, it is for radioSetting array openradio lookup table in radio.c `radio_functions_t dyn_funcs [MAX_RADIOS];`

twatteyne commented on bsp/boards/openmote-b/board_info.h:
```diff
@@ -105,21 +73,21 @@
 #define BSP_ANTENNA_CC2538_24GHZ    GPIO_PIN_4      //!< PD4 -- 2.4ghz
 #define BSP_ANTENNA_AT215_24GHZ     GPIO_PIN_3      //!< PD3 -- subghz
- 
+//#define DAGROOT
```
remove this line

twatteyne commented on bsp/boards/openmote-b/board_info.h:
```diff
@@ -105,21 +73,21 @@
 
 static const uint8_t rreg_uriquery[]        = "h=ucb";
-static const uint8_t infoBoardname[]        = "CC2538";
+static const uint8_t infoBoardname[]        = "openmote-b";
```
"OpenMote B"

twatteyne commented on bsp/boards/openmote-b/board_info.h:
```diff
@@ -105,21 +73,21 @@
 
 //=========================== public ==========================================
-
+void eraseFlash(void);
```
what is this?

twatteyne commented on bsp/boards/openmote-b/cx2538xf53.icf:
```diff
@@ -0,0 +1,109 @@
+/*###ICF### Section handled by ICF editor, don't touch! ****/
```
please explain where this files comes from, and justifty that isn't not an artifact.

minarady1 replied:
this is the linker file necessary for IAR compilation @changtengfei

twatteyne commented on bsp/boards/openmote-b/i2c.c:
```diff
@@ -21,10 +21,10 @@
 
 #define I2C_PERIPHERAL          ( SYS_CTRL_PERIPH_I2C )
 #define I2C_BASE                ( GPIO_B_BASE )
-#define I2C_SCL                 ( GPIO_PIN_3 )
```
why do you have to change I2C for OpenRadio work?!?

minarady1 replied:
Just using updated driver from the subghz project as is (in fact this setting breaks the slot debugpin)

twatteyne commented on bsp/boards/openmote-b/leds.c:
```diff
@@ -59,34 +59,34 @@ uint8_t leds_error_isOn(void) {
     return (uint8_t)(ui32Toggle & BSP_LED_1)>>4;
 }
 
-// orange
```
why do youhave to chnage the LEDs?!?

minarady1 replied:
just used the version from the subghz project as is

twatteyne commented on bsp/boards/openmote-b/radio.c:
```diff
@@ -1,531 +1,325 @@
-#include <source/interrupt.h>
-#include <source/sys_ctrl.h>
-
+    \brief Definition of the Open Radio interface for openmote-b
```
You cannot remove authors, add you name at the bottom of the list.

minarady1 replied:
In fact this a completely new file for the openradio interface, it is just named the same. The original radio.c file  containing cc2538 radio implementation is now this file https://github.com/minarady1/openwsn-fw/blob/develop_FW-866/bsp/chips/cc2538rf/radio_cc2538rf.c

twatteyne commented on bsp/boards/openmote-b/radio.c:
```diff
@@ -1,531 +1,325 @@
+typedef void                (*radio_txNow_cb_t)(void);
+typedef void                (*radio_rxEnable_cb_t)(void);
+typedef void                (*radio_rxNow_cb_t)(void);
+typedef void                (*radio_getReceivedFrame_cb_t)(
```
indentation

twatteyne commented on bsp/boards/openmote-b/radio.c:
```diff
@@ -1,531 +1,325 @@
+typedef struct {
+    radio_reset_cb_t                    radio_reset;
+    radio_init_cb_t                     radio_init;
+    radio_setConfig_cb_t           radio_setConfig;
```
indentation

twatteyne commented on bsp/boards/openmote-b/spi.c:
```diff
@@ -0,0 +1,211 @@
+
+
+//=========================== defines =========================================
+#define SPI_PIN_SSI_CLK             GPIO_PIN_2      //    CLK
```
newline

twatteyne commented on bsp/boards/openmote-b/startup_iar.c:
```diff
@@ -39,8 +39,8 @@
 #define FLASH_START_ADDR                0x00200000
-#define BOOTLOADER_BACKDOOR_ENABLE      0xF7FFFFFF // ENABLED: PORT A, PIN 6, LOW
 #define BOOTLOADER_BACKDOOR_DISABLE     0xEFFFFFFF
+#define BOOTLOADER_BACKDOOR_ENABLE      0xF6FFFFFF // ENABLED: PORT A, PIN 6, LOW
```
why this change?

changtengfei replied:
PA6 is the pin controlling the bootloader backdoor. We probably hadn't figure out the correct configuration at the time we create the openmote-b port.

twatteyne reviewed (changes requested):
I haven't reviewed every file, but the comments in the file I did review are important enough that they require you to rework the code, and restart a new review cycle.

changtengfei commented on bsp/boards/iot-lab_A8-M3/board.c:
```diff
@@ -96,8 +101,48 @@ void board_init(void){
+// used during initialization by sixtop to fire the first sixtop EB
+uint16_t board_getSlotDuration (time_type_t time_type)
+{
+  if (time_type == TIME_MS)
```
Add braces for the if else condition.

changtengfei commented on bsp/boards/iot-lab_M3/board.c:
```diff
@@ -93,8 +98,47 @@ void board_init(void)
+// used during initialization by sixtop to fire the first sixtop EB
+uint16_t board_getSlotDuration (time_type_t time_type)
+{
+  if (time_type == TIME_MS)
```
Braces missing here as well. Move the brace of the function after the parameters. Just be consistent with other functions.

changtengfei commented on bsp/boards/iot-lab_M3/board.c:
```diff
@@ -93,8 +98,47 @@ void board_init(void)
 
+// Getter function for slot_board_vars
+slot_board_vars_t board_getSlotTemplate (void)
+{
```
same here

changtengfei commented on bsp/boards/iot-lab_M3/board.c:
```diff
@@ -93,8 +98,47 @@ void board_init(void)
+
+// Getter function for selected_slot_type
+void board_setSlotType(slotType_t slot_type)
+{
```
same here

changtengfei commented on bsp/chips/at86rf215/radio_at86rf215.c:
```diff
@@ -2,16 +2,17 @@
 
 \author Jonathan Munoz <jonathan.munoz@inria.fr>, July 2016.
-*/
+Modified by Mina Rady <mina1.rady@orange.com>
```
You can write something like this: 

`\author Mina Rady <mina1.rady@orange.com>, April 2020.`

changtengfei commented on openstack/02a-MAClow/IEEE802154E.h:
```diff
@@ -1,347 +1,354 @@
-
-#endif
-
+#ifndef __IEEE802154E_H
```
Do you changed the line ending of this file? The whole file is marked as changed. 
Use the same line ending of the original file.

changtengfei commented on openstack/03a-IPHC/frag.c:
```diff
@@ -337,7 +336,7 @@ void frag_receive(OpenQueueEntry_t *msg) {
                 msg->creator = COMPONENT_FRAG;
                 allocate_vrb(msg, size, tag);
-                return iphc_receive(msg);
+                iphc_receive(msg);
```
Have you confirmed this change with @TimothyClaeys ?

minarady1 replied:
Yes, but it would be good to have another look @TimothyClaeys

minarady1 replied:
yup, but timo can confirm as well @TimothyClaeys

TimothyClaeys replied:
yes, we discussed this earlier.

changtengfei commented on openstack/03a-IPHC/frag.c:
```diff
@@ -436,7 +435,6 @@ static void store_fragment(OpenQueueEntry_t *msg, uint16_t size, uint16_t tag, u
     for (i = 0; i < FRAGMENT_BUFFER_SIZE; i++) {
         if (frag_vars.fragmentBuf[i].datagram_tag == tag && frag_vars.fragmentBuf[i].datagram_offset == offset) {
             openqueue_freePacketBuffer(msg);
-            return;
```
same here

changtengfei replied:
Make sure this is fine since the code will loop though all frag buffer

changtengfei commented on projects/common/01bsp_radio_rx/01bsp_radio_rx.c:
```diff
@@ -1,119 +1,63 @@
 /**
-\brief This program shows the use of the "radio" bsp module.
```
why remove the description?

changtengfei replied:
Is this file still for bsp_radio_rx project?

changtengfei commented on projects/common/01bsp_radio_tx/01bsp_radio_tx.c:
```diff
@@ -1,16 +1,13 @@
 /**
 \brief This program shows the use of the "radio" bsp module.
-
```
keep the newline between paragraphs

changtengfei commented on projects/openmote-b-subghz/03oos_macpong/03oos_macpong.ewt:
```diff
@@ -0,0 +1,2891 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
somehow the .gitignore doesn't work. The file supposes to be not committed.

changtengfei commented on projects/openmote-b/01bsp_openradio_rx/opentestbed_pdr.py:
```diff
@@ -0,0 +1,252 @@
+import paho.mqtt.client as mqtt
```
I guess this file is copied from another port.  If you didn't use it,  don't commit it.

changtengfei reviewed (changes requested):
I left some comments. Some suggestions: 

- for the if/else condition, always use bracket even there is only one line
- the .gitignore somehow doesn't work, some unnecessary files are still committed.

minarady1 committed:
FW-866 fixed slot template timing references as per PR comments

minarady1 committed:
FW-866 merged board_setSlotType (slotType) and slot_board_vars_t board_getSlotTemplate() into    slot_board_vars_t board_selectSlotTemplate(slotType)

minarady1 committed:
FW-866 fixing indentation, variable names for openradio + adjusting 24ghz timing

minarady1 committed:
FW-866 restoring tx/rx common project

minarady1 committed:
increased testbed bootload timeout

minarady1 committed:
fw-866
- openqueue_stats
- increased max neighbors to 45

minarady1 committed:
fw-866
- provided comments on at86rf215 register settings
- updated uinject app with additional parameters
- dynamic eb_portion
- RPL DIO_PERIOD 20 secs

minarady1 committed:
fw-866
- custom ot bootloading capability
- resetStats function rename in ieee154e
- counting ticks for radio in tx and rx states (some inaccuracy in rx to be resolved)
- used 41 slotframe legnth
- disabled fragmentation
